lib/ansible/modules/system/reboot.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'core'}
+
+DOCUMENTATION = r'''
+module: reboot
+short_description: Reboot a \*nix machine
```

Comment:
I believe you don't need `\*`, just `*` in raw-string literal

Reply:
Oh, is that for YAML? It shouldn't need that as well except for the beginning of the value.

Reply:
The only way I could get the docs sanity test to pass was escaping the `*`. I figured having the documentation as a raw string would suffice, but it still failed sanity tests.

Reply:
It's to make the `rst` parser happy. It complains about an unbalanced emphasis character.

Reply:
I would not mention *nix in the short_description, as we want to make this support Windows as well.

Reply:
There is a separate `win_reboot` module.

Reply:
@sivel Sure, but since reboot supports multiple platforms anyway, we might as well make it work on Windows too. That was the original intention anyway, and it helps doing reboots without having to differentiate between platform (i.e. doing them consecutively).

Reply:
That is not the intention of this PR.  In this PR there are 2 separate action plugins/modules, and the `win_reboot` inherits from `reboot` and overrides for it's specific functionality.  `reboot` only supports nix machines.

I'm not the author here of course, so I'll let @samdoran address this further.

Reply:
@sivel I was talking about the original issue and example code. And I am not saying we need this as part of this PR, but if the intention was to add it, we might as well not say *nix in the short_description and provide details in the description.

PS We don't mention *nix in the **user** pr **service** module either.

Reply:
In other modules, we add a reference in the documentation to the Windows counterpart. We should do this here as well (although I would prefer Windows-support in this module at some point ;-))

But again, I don't think we need to add *nix to the short_description.

Reply:
This was parameter-compatible with `win_reboot` at the moment, but I suspect it may not always remain so. For instance, I've had a feature I've wanted to add to win_reboot that optionally checks all the various "do I need a reboot?" flags which would be nonsensical for non-Windows platforms- it's kind of the same reason we split `win_copy` from `copy`... Trying to force these things to be arg/doc compatible will either force lowest-common-denominator behavior or have an increasing number of "this parameter only applies to platform X" kinds of things. It's a tough call, but I'd rather keep them as separate actions with shared impl where appropriate.

Reply:
@nitzmahone Same for Solaris, AIX, FreeBSD, Linux, etc. We already have to provide guidance on how parameters influences each. If you're managing systems in a heterogeneous environment, being able to reboot at the same time is very useful. Having to wait for Windows systems to be reboot first, before you can reboot the Linux systems is just annoying. (And I bet not a lot of people manage heterogeneous environments, so I will be isolated on this one again)

Reply:
The intent was to use `reboot` as a base class that `win_reboot` could subclass since some of the reboot logic needs to be quite a bit different for Windows. Yes, it's not great that you have to do separate tasks to reboot non-homogenous environments, but I wanted to maintain the separation between the two modules to allow them to diverge as needed.

I will remove the reference to `*nix` in the documentation.